### PR TITLE
[WebProfilerBundle] Add channel log filter

### PR DESCRIPTION
**Pull request by ro0NL**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no     
| Deprecations? | no
| Tests pass?   | yes    
| Fixed tickets | #...   
| License       | MIT
| Doc PR        | symfony/symfony-docs#... 

Continuation of #28906

The JS is revised to be more generic;

- support 2 filter types: `level` and `choice` (respectively `Log level` and `Log channel` here)
- remove default filter value support (not used yet, but opportunity kept open) - it requires a bit more work to genericify it.
- filters refines the resultset (e.g. show all logs in the app channel with priority higher than alert)

image

Level filter (works the same as shown in #28906 )

image

Choice filter

image

image

We forgot to update TwigBundle previously, that's still needed after review here.

### CODE REVIEW

File: src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/profiler.css.twig [hunk 1]
```diff
@@ -7,7 +9,7 @@
     --font-sans-serif: 'Helvetica, Arial, sans-serif';
     --page-background: #f9f9f9;
     --color-text: #222;
-    --color-muted: #777;
+    --color-muted: #999;
```
Comment (ro0NL) [inline src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/profiler.css.twig]: this var was overwritten below

File: src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/profiler.css.twig [hunk 2]
```diff
@@ -28,7 +30,6 @@
     --table-header: #e0e0e0;
     --shadow: 0px 0px 1px rgba(128, 128, 128, .2);
     --border: 1px solid #e0e0e0;
-    --color-muted: #999;
```
Comment (ro0NL) [inline src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/profiler.css.twig]: that's here :)

File: src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/profiler.css.twig [hunk 3]
```diff
@@ -892,9 +893,6 @@ tr.status-warning td {
     background-color: var(--base-5);
     color: var(--base-2);
 }
-.tab-content {
```
Comment (ro0NL) [inline src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/profiler.css.twig]: not needed, now specified by `[data-filters]` (the container that requires it)

File: src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/profiler.css.twig [hunk 4]
```diff
@@ -958,45 +991,6 @@ table.logs .metadata {
     display: block;
     font-size: 12px;
 }
-table.logs tr td:last-child { width: 100%; }
```
Comment (ro0NL) [inline src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/profiler.css.twig]: already handled by `.full-width` on the message TH

**Review by ro0NL — COMMENTED** [review]
all good :D and a tiny step closer to sanity.

**Review by fabpot — APPROVED** [review]
/cc @javiereguiluz

**Comment by javiereguiluz** [comment]
Great work Roland! I've tested this in the Symfony Demo app. I only have minor remarks.

**First:** "Ghost filters": when loading the page, filters are displayed by default and they disappear when they shouldn't be displayed. This GIF shows a force reload in the page:

ghost-filters

Maybe it's better to "hide by default" and display when appropriate?

**Second:** "Moving filters":

Right now, the multiple choice filters "move" under some circumstances. See hwen I deselect all of channels and icons disappear:

channel-filter

This kind of movements are distracting and user may click wrongly because of these element displacements. What if we replace the showing/hiding icon by a form checkbox? It doesn't have to be a real form, just a clickable checkbox. Stripe does something similar:

stripe-filter

**Comment by ro0NL** [comment]
I think i've addressed both issues.

image

**Review by fabpot — APPROVED** [review]

**Comment by ro0NL** [comment]
@yceruto a deprecation option doesnt behave correct yet, the opt-out isnt working. Each test triggers still

```
  15x: The option "regions" is deprecated since Symfony 4.2.
    2x in TimezoneTypeTest::testDateTimeZoneInput from Symfony\Component\Form\Tests\Extension\Core\Type
```

**Pull request merged**